client/src/components/DataFiles/DataFilesListing/DataFilesListingCells.jsx

```diff
@@ -3,44 +3,31 @@ import PropTypes from 'prop-types';
 import { Link } from 'react-router-dom';
 import { Checkbox, Icon, LoadingSpinner } from '_common';
 import { Button } from 'reactstrap';
-import { useSelector, useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import './DataFilesListingCells.scss';
 import '../../Onboarding/OnboardingStep.module.scss';
 
+import { useSelectedFiles } from 'hooks/datafiles';
 import formatSize from 'utils/sizeFormat';
 import { formatDateTimeFromValue } from 'utils/timeFormat';
 
 export const CheckboxHeaderCell = () => {
-  const selected = useSelector((state) => state.files.selectAll.FilesListing);
-  const listingLength = useSelector(
-    (state) => state.files.listing.FilesListing.length
-  );
-  const dispatch = useDispatch();
-  const toggleSelect = () => {
-    listingLength &&
-      dispatch({
-        type: 'DATA_FILES_TOGGLE_SELECT_ALL',
-        payload: { section: 'FilesListing' },
-      });
-  };
-  const handleKeyPress = (e) => e.key === ' ' && toggleSelect();
+  const { allSelected, selectAll } = useSelectedFiles();
+  const handleKeyPress = (e) => e.key === 'enter' && selectAll();
   return (
     <Checkbox
-      isChecked={selected}
-      id="FileCheckboxHeader"
-      role="checkbox"
-      aria-label="select all folders and files"
```

Comment:
Any benefit from losing the aria label?

Reply:
Nope, good catch, this probably got mangled during a rebase